nixos/modules/services/torrent/transmission.nix

```diff
@@ -341,8 +341,7 @@ in
     # because BindPaths= needs these directories before.
     system.activationScripts.transmission-daemon = ''
       install -d -m 700 -o '${cfg.user}' -g '${cfg.group}' '${cfg.home}/${settingsDir}'
-    ''
-    + optionalString (cfg.downloadDirPermissions != null) ''
+
```

Comment:
The commands below will fail if `downloadDirPermissions` is set to `null` explicitely. Maybe change the type to `types.str` or keep this conditional?